tooling/template-base/components/LearnerAccess/LoadUserLeaning.tsx

```diff
@@ -0,0 +1,163 @@
+} from '@thoughtindustries/content';
+import { useTranslation } from 'react-i18next';
+import LearnerAccessGridView from './GridView';
+import LearnerAccessListDisplayDropDown from './ListDisplayDropDown';
+import LearnerAccessDisplayListView from './DisplayListView';
+
+const LoadUserLearning = ({ query, kind, sort }: LoadedComponentProps): JSX.Element => {
+  const [gridViewActive, setGridActive] = useState(true);
+
+  // update state to display grid only on mobile
+  const handleResize = () => {
+    if (window.innerWidth < 600) {
+      setGridActive(true);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+  });
+
+  const { data, loading, error } = useUserContentItemsQuery({
+    variables: {
+      query,
+      kind,
+      sort
+    },
+    fetchPolicy: 'network-only'
+  });
+
+  const { i18n } = useTranslation();
+
+  interface ContentUiProps {
+    item: HydratedContentItem;
+    index?: number;
+  }
+
+  const DisplayListView = ({ item }: ContentUiProps) => {
+    // const [listViewDropDown, setListViewDropDown] = useContext(false);
+
+    // item link
+    const hydratedItem = hydrateContent(i18n, item);
+    const itemLink = hydratedItem.href;
+
+    return (
+      // list flex container
+      <LearnerAccessDisplayListView item={item} itemData={data} />
+    );
+  };
+
+  const ListDisplayDropDown = ({ item }: ContentUiProps) => {
+    const { data } = useUserCourseCompletionProgressQuery({
+      variables: {
+        id: item.id
+      }
+    });
+
+    return (
+      // {/* list drop down */}
+      <LearnerAccessListDisplayDropDown item={item} itemData={data} />
+    );
+  };
+
+  const DisplayGridView = ({ item }: ContentUiProps) => {
+    // item link
+    const hydratedItem = hydrateContent(i18n, item);
+    const itemLink = hydratedItem.href;
+
+    return (
+      // {/* course card */}
+      <LearnerAccessGridView item={item} itemUrl={itemLink} />
+    );
+  };
+
+  if (loading) return <LoadingDots />;
+
+  if (error) return <>{error.message}</>;
+
+  if (!data || !data.UserContentItems) return <></>;
+
+  return (
+    <>
+      {/* grid/list toggle */}
+      <div className="hidden sm:flex justify-end p-3">
+        {/* list display button */}
+        {gridViewActive ? (
+          <button
+            className="flex border rounded-l-md w-9 h-9 place-content-center items-center"
+            onClick={() => setGridActive(false)}
+          >
+            <img src={listViewSelector} />
+          </button>
+        ) : (
+          <button
+            className="flex border rounded-l-md w-9 h-9 place-content-center items-center bg-blue-600"
+            onClick={() => setGridActive(false)}
+          >
+            <img src={listViewSelector} />
+          </button>
+        )}
+        {/* grid display button */}
+        {gridViewActive ? (
+          <button
+            className="flex border rounded-r-md w-9 h-9 place-content-center items-center bg-blue-600"
```

Comment:
Seems like either this or line 123 should have `setGridActive` as `false`?

Reply:
setGridActive is called in the child components (line 75 for example), would you like this changed?

Reply:
@AlexONeillTI Ah, no I misunderstood the intent here. Since it seems like the only change in the element is the classes, think we can just have a ternary for the classnames instead of a whole element, might make it a bit more readable.

Reply:
solution pushed